python/lib/dependabot/python/file_updater/pipfile_file_updater.rb

```diff
@@ -15,73 +15,115 @@ module Dependabot
   module Python
     class FileUpdater
       class PipfileFileUpdater
+        extend T::Sig
         require_relative "pipfile_preparer"
         require_relative "pipfile_manifest_updater"
         require_relative "setup_file_sanitizer"
 
-        DEPENDENCY_TYPES = %w(packages dev-packages).freeze
+        DEPENDENCY_TYPES = T.let(%w(packages dev-packages).freeze, T::Array[String])
 
+        sig { returns(T::Array[Dependabot::Dependency]) }
         attr_reader :dependencies
+
+        sig { returns(T::Array[Dependabot::DependencyFile]) }
         attr_reader :dependency_files
+
+        sig { returns(T::Array[Dependabot::Credential]) }
         attr_reader :credentials
+
+        sig { returns(T.nilable(String)) }
         attr_reader :repo_contents_path
 
+        # rubocop:disable Metrics/AbcSize
+        sig do
+          params(
+            dependencies: T::Array[Dependabot::Dependency],
+            dependency_files: T::Array[Dependabot::DependencyFile],
+            credentials: T::Array[Dependabot::Credential],
+            repo_contents_path: T.nilable(String)
+          ).void
+        end
         def initialize(dependencies:, dependency_files:, credentials:, repo_contents_path:)
           @dependencies = dependencies
           @dependency_files = dependency_files
           @credentials = credentials
           @repo_contents_path = repo_contents_path
-        end
-
+          @updated_pipfile_content = T.let(nil, T.nilable(String))
+          @updated_lockfile_content = T.let(nil, T.nilable(String))
+          @updated_generated_files = T.let(nil, T.nilable(T::Hash[Symbol, String]))
+          @pipfile = T.let(nil, T.nilable(Dependabot::DependencyFile))
+          @lockfile = T.let(nil, T.nilable(Dependabot::DependencyFile))
+          @setup_files = T.let(nil, T.nilable(T::Array[Dependabot::DependencyFile]))
+          @setup_cfg_files = T.let(nil, T.nilable(T::Array[Dependabot::DependencyFile]))
+          @requirements_files = T.let(nil, T.nilable(T::Array[Dependabot::DependencyFile]))
+          @updated_dependency_files = T.let(nil, T.nilable(T::Array[Dependabot::DependencyFile]))
+          @updated_pipfile_content = T.let(nil, T.nilable(String))
+          @parsed_lockfile = T.let(nil, T.nilable(T::Hash[String, T::Hash[T.untyped, T.untyped]]))
```

Comment:
Is it possible to define a better type than `T::Hash[String, T::Hash[T.untyped, T.untyped]`. Even just a better key value for the 2nd `T::Hash`.